src/Sulu/Bundle/EventLogBundle/Tests/Functional/Infrastructure/Doctrine/Repository/EventRecordRepositoryTest.php

```diff
@@ -95,10 +95,31 @@ public function testAddAndCommit(): void
         $eventRecord = $this->repository->createForDomainEvent($this->domainEvent->reveal());
         static::assertCount(0, $entityRepository->findAll());
 
-        $this->repository->add($eventRecord);
+        $this->repository->addAndCommit($eventRecord);
+        $eventRecords = $entityRepository->findAll();
+        static::assertCount(1, $eventRecords);
+        static::assertNull($eventRecords[0]->getEventPayload());
+
+        $this->repository->addAndCommit($eventRecord);
+        static::assertCount(2, $entityRepository->findAll());
+    }
+
+    public function testAddAndCommitWithPayload(): void
+    {
+        static::bootKernel(['environment' => 'with_payload']);
+        $this->setUp();
```

Comment:
not really happy about this, the other option would be to create a separate test-class for `with_payload` tests